db/benchmark.js

```diff
@@ -0,0 +1,58 @@
+const db = require('./mongodb');
+
+const testMongoDB = async (totRuns = 1) => {
+  for (let i = 0; i < totRuns; i += 1) {
+    const needle = i < 9
+      ? (i * 1e6) + Math.floor(1e6 * Math.random())
+      : Math.floor(10e6 * Math.random());
+
+    console.log('Running #', i);
+    console.log('  Searching for:', needle);
+
+    console.time('findByRestaurantId');
+    const data = await db.findOneByRestaurantId(needle);
+    console.timeEnd('findByRestaurantId');
+
+    console.log('  Found:', data.restaurantName);
+  }
+  if (totRuns > 0) {
+    db.mongoose.disconnect();
+  }
+};
+
+const testMongoDBName = async (needles) => {
+  for (let i = 0; i < needles.length; i += 1) {
+    const needle = needles[i];
+
+    console.log('Running #', i);
+    console.log('  Searching for:', needle);
+
+    console.time('findByRestaurantId');
+    const data = await db.Restaurant.findOne({ restaurantName: needle });
+    console.timeEnd('findByRestaurantId');
+
+    console.log('  Found:', data.restaurantName);
+  }
+
+  db.mongoose.disconnect();
+};
+
+
+// testMongoDB(10);
+
+testMongoDB(0);
```

Comment:
As far as I can tell everything looks good. Just something that confused me on this part: why is a value passed into testMongoDB on this invocation when "(totRuns = 1)" is supplied in the function definition?

Reply:
oh the totRuns = 1 is just a default value in case something isn't supplied, but most of the time you'll want to run the benchmark more than once.